crates/anvil/core/src/eth/transaction/mod.rs

```diff
@@ -138,7 +138,7 @@ pub fn transaction_request_to_typed(
                 gas_limit: gas.unwrap_or_default(),
                 value: value.unwrap_or(U256::ZERO),
                 input: input.into_input().unwrap_or_default(),
-                to: match to {
+                to: match to.unwrap_or(TxKind::Create) {
```

Comment:
I believe this should only accept `TxKind::To`, ref https://github.com/alloy-rs/alloy/pull/355